### PR TITLE
manipulate_wp: massive overhall from the german open 2024

### PR DESCRIPTION
**Pull request by techtasie**
We changed the MP to now use the center of the laser-line distance into respect to the ideal machine position.

### CODE REVIEW

**Comment by MatteoTschesche** [comment]
You should mention that laser-lines using fawkes is not supported anymore.

**Review by MatteoTschesche — CHANGES_REQUESTED** [review]
Thanks for the work, here is some feedback.

File: src/lua/skills/robotino/manipulate_wp.lua [hunk 1]
```diff
@@ -426,6 +435,9 @@ function INIT:init()
       fsm.vars.c = "C3"
     end
   end
+  if fsm.vars.map_pos ~= false then
```
[the remark this answers is not in the document]
Comment (techtasie) [reply src/lua/skills/robotino/manipulate_wp.lua:438]: okay i will extract out all safe put stuff bit hard to seperate but you have a fair point.

File: src/lua/skills/robotino/manipulate_wp.lua [hunk 2]
```diff
@@ -133,47 +121,60 @@ if config:exists("plugins/object_tracking/shelf_values/right_shelf_offset_side")
...
+      local robot_center =  tfm.transform6D({
+        x=line_center.x,
+        y=line_center.y,
+        z=0,
+        ori = fawkes.tf.create_quaternion_from_yaw(line:bearing())},line:frame_id(), "base_link")
+      local robot_distance = math.sqrt(robot_center.x * robot_center.x + robot_center.y * robot_center.y)
+
+       if (fsm.vars.map_pos == true and distance < 0.50) or fsm.vars.map_pos == false then
+          if best_distance == nil then
+            best_distance = distance
```
[the remark this answers is not in the document]
Comment (techtasie) [reply src/lua/skills/robotino/manipulate_wp.lua]: Thank you!

**Review by MatteoTschesche — APPROVED** [review]
Nice work, thanks!

**Pull request merged**